src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -25,31 +25,30 @@ public partial class RegexGenerator : IIncrementalGenerator
     {
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
+            // To avoid invalidating the generator's output when anything from the compilation
+            // changes, we will extract from it the only thing we care about: whether unsafe
+            // code is allowed.
+            IncrementalValueProvider<bool> allowUnsafeProvider =
+                context.CompilationProvider
+                .Select((x, _) => x.Options is CSharpCompilationOptions { AllowUnsafe: true });
+
             // Contains one entry per regex method, either the generated code for that regex method,
             // a diagnostic to fail with, or null if no action should be taken for that regex.
             IncrementalValueProvider<ImmutableArray<object?>> codeOrDiagnostics =
                 context.SyntaxProvider
 
-                // Find all MethodDeclarationSyntax nodes attributed with RegexGenerator
-                .CreateSyntaxProvider(static (s, _) => IsSyntaxTargetForGeneration(s), static (ctx, _) => GetSemanticTargetForGeneration(ctx))
+                // Find all MethodDeclarationSyntax nodes attributed with RegexGenerator and gather the required information
+                .CreateSyntaxProvider(IsSyntaxTargetForGeneration, GetSemanticTargetForGeneration)
                 .Where(static m => m is not null)
 
-                // Pair each with the compilation
-                .Combine(context.CompilationProvider)
-
-                // Use a custom comparer that ignores the compilation. We want to avoid regenerating for regex methods
-                // that haven't been changed, but any change to a regex method will change the Compilation, so we ignore
-                // the Compilation for purposes of caching.
-                .WithComparer(new LambdaComparer<(MethodDeclarationSyntax?, Compilation)>(
-                    static (left, right) => EqualityComparer<MethodDeclarationSyntax>.Default.Equals(left.Item1, right.Item1),
-                    static o => o.Item1?.GetHashCode() ?? 0))
+                // Pair each with whether unsafe code is allowed
+                .Combine(allowUnsafeProvider)
 
                 // Get the resulting code string or error Diagnostic for each MethodDeclarationSyntax/Compilation pair
```

Comment:
This comment is out-of-date now, right?  It's no longer a MethodDeclarationSyntax/Compilation pair?

Reply:
Fixed.